### DISCUSSION TITLE
Set the texture name

### DISCUSSION THREAD

**Discussion by slightHYP** [google/model-viewer#3132] [Q&A] [answered]
Hi, I was wondering if there is the possibility to set a custom name (or id, an unique value btw) to a dynamic generated texture.

I have something like this (similar to this [example](https://modelviewer.dev/examples/scenegraph/#createTexturesExample))

```
// Creates a new texture.
let texture = await this._modelViewerElement.createTexture(textureUrl, mimeType);
// Apply the new texture to the specified channel.
materialChannel.setTexture(texture);
```
My idea is to set a specific name (id or something else) to the texture object in order to have an easy comparison value when I'm going to request the texture change from the interface.

Currently I compare the source.uri of the texture applied to a specific channel with the new one coming from the UI, not the best solution. This in not optimized, moreover it could introduce issues depending the possible heterogenous sources set in the frontend.

I noticed that there is the `get name()` function both `Texture.ts` and in `Image.ts` where the name is hardcoded (`'adhoc_image'`) in the constructor.
Is there any reason why there is not a setter function for the name? Thank you in advanced

**Comment by samaneh-kazemi** [comment, marked as answer]
@elalish Any inputs?

**Reply by elalish** [reply]
@slightHYP That seems like a reasonable request. Would you mind opening a PR to add it?

**Reply by slightHYP** [reply]
@samaneh-kazemi @elalish thanks for your feedback. Sure I will open a PR as soon as possible linking this discussion.

**Reply by slightHYP** [reply]
PR opened #3175
